client/lib/interval/runner.js

```diff
@@ -32,31 +37,27 @@
  *  which is a safe function to call at any time, meaning that it
  *  won't overlap timers or break things if we called it needlessly.
  */
-
-import { fromJS } from 'immutable';
-
 export const EVERY_SECOND = 1000;
 export const EVERY_FIVE_SECONDS = 5 * 1000;
 export const EVERY_TEN_SECONDS = 10 * 1000;
 export const EVERY_THIRTY_SECONDS = 30 * 1000;
 export const EVERY_MINUTE = 60 * 1000;
 
-const initialState = fromJS( {
+const initialState = {
 	nextId: 1,
 	periodTimers: {
-		EVERY_SECOND: null,
-		EVERY_FIVE_SECONDS: null,
-		EVERY_TEN_SECONDS: null,
-		EVERY_THIRTY_SECONDS: null,
-		EVERY_MINUTE: null
+		[ EVERY_SECOND ]: null,
+		[ EVERY_FIVE_SECONDS ]: null,
+		[ EVERY_TEN_SECONDS ]: null,
+		[ EVERY_THIRTY_SECONDS ]: null,
+		[ EVERY_MINUTE ]: null,
 	},
-	actions: []
-} );
+	actions: [],
+};
 let state = initialState;
 
-const increment = a => a + 1;
-const addToList = item => list => list.push( item );
-const removeFromList = id => list => list.filterNot( o => o.get( 'id' ) === id );
+const addToList = item => list => [ ...list, item ];
+const removeFromList = id => list => reject( list, { id } );
```

Comment:
Nice! Although is there any reason not to just go ahead and jump to a native `Map` here?

Reply:
I thought about it, but decided i wanted to be true to the original intent of the code. My thought was that it would be easier to keep the tests working if i kept state-as-immutable. I think it paid off. We could flip things around in a future PR.

Reply:
why did they not make them immutable 🤦‍♂️

Reply:
ikr